pkg/typesystem/typesystem.go

```diff
@@ -1066,11 +1078,35 @@ func (t *TypeSystem) validateTypeRestrictions(objectType string, relationName st
 				}
 			}
 		}
+
+		if related.Condition != "" {
+			// Validate the conditions referenced by the relations are included in the model.
+			if _, ok := t.conditions[related.Condition]; !ok {
+				return ErrConditionUndefined
+			}
+		}
 	}
 
 	return nil
 }
 
+// validateConditions validates the conditions provided in the model.
+//  1. Each condition should define a well-formed expression,
+//  2. Its parameters should be well-defined
+//  3. The expression and parameters should be compilable.
+func (t *TypeSystem) validateConditions() error {
+	for key, c := range t.conditions {
+		if key != c.Name {
+			return fmt.Errorf("condition key '%s' does not match condition name '%s'", key, c.Name)
+		}
```

Comment:
when would this happen? also, this isn't covered with tests and i'm not sure how you would write a test for it. i suggest removal

Reply:
this protects agains the following:

```
map[string]*openfgav1.Condition{
	"condition1": {
		Name:       "condition2",
		Expression: "param1 == 'ok'",
		Parameters: map[string]*openfgav1.ConditionParamTypeRef{
			"param1": {
				TypeName: openfgav1.ConditionParamTypeRef_TYPE_NAME_STRING,
			},
		},
	},
}
```